### PR TITLE
feat(mobile): add a spray wall from a photo (SW-09)

### LINKED ISSUE #5442: SW-09: Mobile: add-a-wall flow (capture, anchors, detect, review, publish)
Epic #5346 (SW-09). The "Add a spray wall" flow: name it, pick a photo, optionally mark anchors, upload, detect holds on the phone, review them in the editor, publish.

## Why

This is the front door of the feature. It is flag-gated so it can ship dark, and it must work with the photo library alone (the camera button and on-device detection are fingerprint-gated on SW-02) and with zero detections (the SW-08 editor in manual mode), so a model failure never blocks a user.

## Scope

- Entry: a tile on `packages/mobile/app/boards/index.tsx` next to the "Create board" `BoardModeCard` → `router.push('/boards/spray/new')`; `spray/*` registered in `app/boards/_layout.tsx` as pushed screens (routes, not sheets: `docs/mobile-sheets-vs-routes.md` rule 3, full-screen pan / pinch). Gated by the mobile flag `spray-walls` in `FEATURE_FLAG_DEFINITIONS` (`packages/mobile/src/providers/feature-flags-provider.tsx`), off when unresolved, and by `climbCreation` for spray.
- `app/boards/spray/new.tsx`, an in-screen stepper:
  1. Meta: extract `BoardMetaFields.tsx` from `BoardForm.tsx`'s "More options" (name, isPublic / isUnlisted / hideLocation, location via `useDeviceLocation`, gym via `GymPickerSheet`), angle typed by the owner, `use-spray-wall-builder.ts` = the meta half of `use-board-builder.ts` with no config cascade.
  2. Photo: library pick (`ImagePicker.launchImageLibraryAsync`); "Take photo" (`launchCameraAsync`) shown only on binaries whose fingerprint includes SW-02; `compressPickedImage` (`src/lib/image-compression.ts`) to 2048 px long edge JPEG q0.85, which also converts HEIC and bakes EXIF orientation.
  3. Anchors (optional, "Skip" default): `SprayCornerMarker.tsx`, four draggable handles (Reanimated + RNGH), guidance copy "shoot at 1×, stand square to the wall". No width / height fields: the wall's frame is the photo (owner decision 2026-09-14).
  4. Upload with progress: `upload-image.ts` multipart (keeps the SDK-57 `File.bytes()` workaround; release builds pin RN fetch), expo-file-system upload task for progress with an indeterminate fallback → `createSprayWall` + `createSprayWallVersion`.
  5. Detection: `ensureModel()` (SW-02) → `@boardsesh/hold-detection` over the compressed photo with per-tile progress → candidates in **photo px** mapped to canonical through the version's homography; model unavailable or `null` → skip straight to the editor in manual mode.
  6. Review: SW-08 editor on the spray target with the candidates loaded.
  7. Publish: `publishSprayWallVersion` → `registerSprayWall` → `useActivateBoard` → Climbs tab with an empty state that says what to do next ("Set the first climb").
- BLE: never expose an LED toggle for a wall; `createSprayWall` writes `has_leds=false` and the mobile flow must not offer `hasLeds` (the BLE suppression is per-row data, not per board type: `scanFamilyForBoard('spray')` would return `aurora` and encode against empty LED placements — #5453 review, finding 5). A capability-derived guard (`getBoardCapabilities('spray').nativeBoardControl === false` short-circuiting the scan) is welcome here as a one-line defence in depth; it is not BLE protocol code but still gets a Fable/Astra read.
- Errors: upload failure retries the upload step only; detection failure falls back to manual with a toast; caps (10 walls) surface as copy, not as a crash.
- i18n in all four locales (`boards.json`, glossaries for es / fr / de); Velvet Send tokens; performance checklist.

## Out of scope

Reset (SW-13), sharing UI beyond the visibility fields (SW-14), offline (SW-15).

## Acceptance

- QA plan a phone tester can follow: pick a photo → upload progress → detection progress → review (pinch / zoom, add, drag, resize, delete, slider, undo) → Done → the wall appears in My Boards with its photo → Climbs tab shows the empty state.
- Same flow with the model download blocked (airplane mode after login) lands in manual mode and still publishes.
- Same flow on a binary without SW-02 shows no camera button and still publishes.
- Telemetry (SW-17 names) emitted: `Spray Wall Photo Picked`, `Spray Wall Upload Finished`, `Spray Wall Detection Finished`, `Spray Holds Reviewed`, `Board Created` with `boardType: 'spray'`.
- Ships by OTA; no native change.

## Where

`packages/mobile/app/boards/spray/new.tsx` (new), `app/boards/_layout.tsx`, `app/boards/index.tsx`, `packages/mobile/src/components/board-discovery/BoardMetaFields.tsx` (new, extracted), `use-spray-wall-builder.ts` (new), `SprayCornerMarker.tsx` (new), `packages/mobile/src/lib/upload-image.ts`, `src/lib/image-compression.ts`, `src/providers/feature-flags-provider.tsx`, `packages/shared/i18n/locales/*/boards.json`.

## Blocked by / Blocks

- Blocked by: SW-05 (#5438), SW-06 (#5439), SW-08 (#5441); SW-02 (#5435) for the camera button and on-device detection only.
- Blocks: SW-13 (#5446).

## Size

L.

### PR DESCRIPTION
**Pull request by marcodejongh**
## Summary

The front door for spray walls (epic #5346, SW-09). A tile next to "Create board" on the boards picker opens `/boards/spray/new`, an in-screen stepper that takes a wall from a photograph to a board you can set climbs on.

Seven steps, one route: **meta → photo → anchors → upload → detect → review → publish**. It is a route and not a sheet because two of its steps — the corner markers and the SW-08 hold editor — are full-screen pan-and-pinch surfaces (`docs/mobile-sheets-vs-routes.md`, rule 3).

What each step does:

- **Meta** — name, gym, visibility, location and the angle the owner measured. The name/gym/visibility/location block is extracted out of `BoardForm`'s "More options" into `BoardMetaFields.tsx` and used by both forms, so there is one copy rather than two. `use-spray-wall-builder.ts` is the meta half of `use-board-builder.ts` with no config cascade: a wall's layout and size are allocated server-side, and its holds come from the photo. Every visibility default is inverted from a catalogue board's — a wall is somebody's home.
- **Photo** — library pick first. "Take a photo" appears only on a binary whose Info.plist declares the camera, which is a check on `Application.nativeApplicationVersion` (≥ 2.6.0, SW-02's train) and not on a build flag: iOS *terminates* rather than denies when the usage description is missing, and this slice rides an OTA into 2.5.x phones. `compressPickedImage` makes a 2048 px q0.85 JPEG, which also converts HEIC and bakes the EXIF orientation into the pixels — the anchors and the homography both live in that pixel space.
- **Anchors** — optional, Skip is the default, four Reanimated/RNGH handles that only cross to JS when a finger lifts. A quad that crosses itself is refused in the reducer, because the backend's fallback for a degenerate quad is the identity matrix: an un-rejected bow-tie does not fail loudly, it silently puts every hold somewhere plausible and wrong.
- **Upload** — `createSprayWall` (the photo handler authorises against a wall the caller owns, so the wall has to exist first), then a multipart POST with real byte progress via the `expo-file-system` upload task and an indeterminate fallback, then `createSprayWallVersion`. The anchors are rescaled onto the photo the server actually stored.
- **Detect** — the on-device detector is *registered*, not imported: SW-02 is the epic's one native PR and is not in this branch's ancestry, so nothing here pulls `onnxruntime-react-native` into the bundle. With no detector the step answers "unavailable" and the flow goes straight to the editor in manual mode, which is the path the epic decided the product lives on either way.
- **Review** — SW-08's `SprayHoldEditorScreen` with the candidates, in photo pixels, rescaled from the local file onto the stored one.
- **Publish** — `publishSprayWallVersion`, then SW-07's `invalidateSprayWallRenderData` so the device that published re-registers immediately, then `useActivateBoard` and back to Climbs.

Everything that can fail, fails somewhere the climber can act. A failed upload retries the **upload**, against the wall that already exists — and a retry once a draft exists goes to the editor rather than re-uploading into `createSprayWallVersion`'s "already has an unfinished photo". Publishing and binding the wall are latched apart, so a failed bind retries the bind and never re-publishes.

Leaving mid-flow is not a failure at all, and the copy that says so is now true: the flow opens by asking `mySprayWalls` for a wall of the caller's own with no published version and, if it finds one, offers "Pick up where you left off" (rejoining at the editor when the draft already carries a photo, otherwise at the photo step, reusing the wall either way) or "Start over" (discard the draft, delete the empty wall). Without that, every abandoned run left an orphan row against a cap of ten.

Behind the mobile flag `spray-walls`, added to `FEATURE_FLAG_DEFINITIONS`. A positive rollout flag read as `=== true`, so unresolved, absent and off all hide it — the route redirects to the picker rather than trusting the tile to be the only door.

Author-side verification: `vp run typecheck:mobile` clean; `vp run test:mobile` 926 files / 10,352 tests green; `vp run check:mobile-bundle` OK; `vp check` 0 errors. Both guards were mutation-tested — flipping the flag gate to `!== false` fails 2 cases, removing the convex-quad rejection fails 1.

JS-only. No `app.config.ts`, `plugins/`, `modules/`, `patches/` or native dependency: the fingerprint does not move and this ships by OTA.

Known gap, tracked separately: a wall resumed from an earlier sitting always publishes **private**, whatever visibility the climber picked the first time — the choice only ever lives in React state and `createSprayWall` deliberately forces private, so it cannot be recovered here. Fails in the safe direction and is fixable from board edit. Needs persisted intent at draft time: #5513.

Stacked on #5481 (SW-08) → #5479 (SW-07); merge those first, then this PR retargets to main.

## Release Notes

Photograph your spray wall, let the app find the holds, fix what it missed, and start setting. Name the wall, mark its corners if you shot it off-square, and it lands in your boards ready for the first climb. Behind a flag while we roll it out.

## Test plan

1. Profile → More → Feature Flags → enable "Spray walls".
2. Boards tab → a "Spray wall" tile appears; tap it.
3. Name the wall, drag the angle slider, choose a photo.
4. Drag the four corner rings → tap "Use these corners".
5. Add a hold, Save, Done → the wall becomes active.

## Risk

Risk: 3/5 — new flag-gated flow; no existing screen changes behaviour

Closes #5442

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Q7BjHrRALJ1gjtiRWm5rLV

### CODE REVIEW

File: packages/mobile/src/components/board-discovery/use-spray-wall-builder.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  angle?: number;
+  isPublic?: boolean;
+  isUnlisted?: boolean;
+  hideLocation?: boolean;
+  locationName?: string;
+};
+
+/** True when `value` is an angle a wall can actually be built at. */
+export function isValidSprayAngle(value: number): boolean {
+  return Number.isFinite(value) && Number.isInteger(value) && value >= MIN_SPRAY_ANGLE && value <= MAX_SPRAY_ANGLE;
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/src/components/board-discovery/use-spray-wall-builder.ts]: Fixed: the angle is now the shared `SPRAY_ANGLES` list on the board builder's own snapping slider rather than free text, so 37 has nowhere to live — `use-spray-wall-builder.test.ts` pins `SPRAY_ANGLE_OPTIONS` against `SPRAY_ANGLES`, rejects 37/42/71/12.5, and checks a seeded off-list angle falls back to the default.

File: packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+  // ============================================
+  // Step 0 — is there a wall to pick up?
+  // ============================================
+
+  // Only while the flow is actually asking. Once it has an answer the query is
+  // dead weight, and refetching it mid-flow could offer to resume the very wall
+  // this run just created.
+  const mySprayWalls = useMySprayWalls({ enabled: state.step === 'resuming' });
+  const walls = mySprayWalls.data;
+  const wallsSettled = !mySprayWalls.isPending;
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx]: Fixed: the resume check now waits for a fetch that SETTLED after this mount (`!isFetching && max(dataUpdatedAt, errorUpdatedAt) > mountedAt`) rather than for `isPending`, so the cached pre-creation list can no longer latch the prompt away — `resume-draft.test.ts` covers the selection and `add-wall-machine.test.ts` the `hasUnfinishedWall` / resume transitions.

File: packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,847 @@
...
+      <View style={styles.flex}>
+        <View style={[styles.reviewBar, { borderBottomColor: systemColors.separator }]}>
+          <Text variant="footnote" color={systemColors.secondaryLabel} style={styles.reviewHint}>
+            {detectionSummary(state, t)}
+          </Text>
+          <Button
+            title={t('sprayWizard.review.done')}
+            variant="filled"
+            onPress={() => dispatch({ type: 'REVIEW_DONE' })}
+            disabled={!state.hasSavedHolds}
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx:616]: Fixed: `resumeTargetFor` now carries the draft's `addedHoldCount` and `RESUMED_AT_REVIEW` latches `hasSavedHolds` from it, so a resumed draft with saved holds can publish without a gratuitous edit — tested both ways round in `add-wall-machine.test.ts` ('unlocks Done without a gratuitous edit' / 'leaves Done locked for a draft with no holds on it yet') and in `resume-draft.test.ts`.

File: packages/mobile/src/components/board-discovery/use-spray-wall-builder.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+   * moves.
+   */
+  const buildCreateInput = useCallback((): CreateSprayWallInput | null => {
+    const trimmedName = name.trim().slice(0, MAX_NAME_LENGTH);
+    if (trimmedName.length === 0 || angle == null) return null;
+    return {
+      name: trimmedName,
+      angle,
+      isPublic,
+      isUnlisted,
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/src/components/board-discovery/use-spray-wall-builder.ts]: Fixed: `buildCreateInput` now always sends `isPublic: false, isUnlisted: false`, and the chosen visibility is remembered in `pendingVisibility()` and applied by `updateSprayWall` immediately after the first publish (after the `PUBLISHED` latch, so a bind retry re-applies rather than re-publishes) — `use-spray-wall-builder.test.ts` covers the private creation, the remembered choice, and the no-op for a wall left private.

File: packages/mobile/app/boards/_layout.tsx [hunk 5]
```diff
@@ -38,6 +38,12 @@ export default function BoardsLayout() {
           unfollowing live on the picker's cards (#4623). */}
       <Stack.Screen name="manage" options={{ title: t('myBoards.title') }} />
       <Stack.Screen name="edit" options={{ title: tBoards('mobile.edit.screenTitle') }} />
+      {/* The add-a-wall flow, pushed like the builder above it. A ROUTE and not a
+          sheet: two of its steps (the corner markers and the hold editor) are
+          full-screen pan-and-pinch surfaces, which `docs/mobile-sheets-vs-routes.md`
+          rule 3 keeps off a sheet's own drag. Flag-gated inside the screen — the
+          route existing is not the same as the feature being reachable. */}
+      <Stack.Screen name="spray/new" options={{ title: tBoards('sprayWizard.screenTitle') }} />
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/app/boards/_layout.tsx:46]: Fixed: the screen now registers a `beforeRemove` listener gated on the same `shouldConfirmLeave(state)` predicate the footer uses (`isBusy || leavingKeepsDraft`), showing the same leave sheet and re-dispatching the removal only on confirm — the predicate is tested in `add-wall-machine.test.ts` ('shouldConfirmLeave': quiet before any write, asks during upload/detection/publish, asks once a draft exists, stops once published).

File: packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,847 @@
...
+      return;
+    }
+
+    const resume = async () => {
+      // The list carries no version history, so the draft — and whether it has a
+      // photo — takes one more round trip. A failure here rejoins at the photo
+      // step, which is the safe half: the wall is reused either way.
+      const full = await fetchSprayWallVersions(resumable.uuid).catch(() => null);
+      if (full?.board) boardRef.current = full.board;
+      const target = resumeTargetFor(resumable, full?.versions ?? []);
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx]: Fixed: a failed version-history request no longer collapses to `null`; it surfaces `sprayWizard.resume.checkFailed` on the resuming step with a retry that re-runs both the list and the history, so a wall with an open draft can never be mistaken for a bare one — the 'rejoins at the photo step' cases in `resume-draft.test.ts` now only describe a genuinely empty history.

File: packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,847 @@
...
+      // The anchors were tapped on the LOCAL file. The stored object is the
+      // server's own re-encode of it, so the quad is carried across by the same
+      // ratio the candidates are.
+      const anchors = state.anchors
+        ? state.anchors.map((point) => rescalePoint(point, { width: photo.width, height: photo.height }, stored))
+        : null;
+
+      const version = await createVersionAsync({ wallUuid: wall.wallUuid, photoId: uploaded.photoId, anchors });
+      const draft: CreatedWallDraft = { ...wall, versionId: version.id, versionNumber: version.number };
+      dispatch({ type: 'DRAFT_CREATED', draft });
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/src/components/spray-wall/SprayWallWizardScreen.tsx:412]: Fixed: an upload retry against a wall that already exists first re-reads its versions and, if an open draft with a photo is there, adopts it via `RESUMED_AT_REVIEW` instead of uploading a second photo into the one-draft refusal — the adoption path is the same `resumeTargetFor` covered in `resume-draft.test.ts`, and it also spares the private bucket the orphaned copy.

File: packages/mobile/app/boards/spray/new.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import { resolveBoardReturnTo } from '../../../src/lib/boards/board-return-to';
+
+export default function NewSprayWall() {
+  const params = useLocalSearchParams<{ returnTo?: string }>();
+  const enabled = useSprayWallsEnabled();
+
+  // Unresolved flags read as off (`useSprayWallsEnabled`), so this also covers
+  // the first frames of a cold open on a deep link: back to the picker, which is
+  // a real screen, rather than a blank one.
+  if (!enabled) return <Redirect href="/boards" />;
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/mobile/app/boards/spray/new.tsx:29]: Fixed: `FeatureFlagsProvider` now exposes `useFeatureFlagsResolved()` (true once PostHog answers, once a static bag is supplied, or after a 2 s backstop) and the route renders nothing until it is true, so a cold deep link is no longer redirected on the empty first frame — tested in `spray-walls-flag.test.tsx` ('is false on the first frame with nothing supplied' / 'is true immediately for a statically supplied bag').

**Comment by marcodejongh** [comment]
## Review items addressed

Every distinct item across the seven claude[bot] reviews, with where it landed. Code was pushed: head is now `3473acd7f`.

### Fixed in `3473acd7f` (this round)

- [x] **Upload retry could loop on a failed version check** (`SprayWallWizardScreen.tsx`). Correct, and the Codex round did *not* cover it — that round added the reconcile query, this one covers the query itself failing. Extracted to `planUploadRetry` (`resume-draft.ts`) with three answers: `adopt` a draft a lost response already created, `upload` when the wall genuinely has nothing, `blocked` when the wall could not be *read*. `blocked` surfaces `sprayWizard.upload.checkFailed` instead of re-uploading into a guaranteed refusal. Tested (4 cases) and mutation-tested — flipping `blocked` back to `upload` fails one.
- [x] **`useSprayWallBuilder` return not memoised** (CLAUDE.md mobile checklist). Wrapped in `useMemo`. Two tests: stable identity across a re-render, new identity when a held value changes.
- [x] **`BACK_TARGET.detect` hazard**. Entry removed. Backing into `photo` with a draft open would hit `runUpload`'s draft guard and sit there doing nothing — worse than no way back. `upload` keeps its way back, because a draft cannot exist on that step. Tested.
- [x] **`findResumableWall` picks by server order**. The comment was wrong about recency: `mySprayWalls` orders `spray_walls.created_at` DESC (`resolvers/board/spray-walls.ts`), so the first unfinished wall *is* the most recent. Comment corrected to cite the ordering; no client-side re-sort, because the wall payload carries no timestamp to sort by.
- [x] **Missing test for the null-history retry path**. That is the `planUploadRetry` suite above — a pure helper rather than a screen test, so the branch is pinned rather than merely exercised.

### Already fixed at `f01cf4751` or earlier

- [x] **i18n `{{value}}` vs `{{count}}`** — `detectionSummary` passes `{ value }`; `review-summary-copy.test.ts` renders the real catalog in all four locales and one case pins that the wrong name leaves the placeholder, so the assertion is not vacuous.
- [x] **`SprayCornerMarker` handles frozen after Clear** — effect keyed on the seed's values writes the shared values back.
- [x] **`parseVersion` dead `length === 0` branch** — removed.
- [x] **Division by zero / NaN in the preview height** — `previewHeight()` guards both dimensions and falls back to 4:3.
- [x] **`docs/spray-walls.md` said the UI did not exist** — new "Adding a wall" section (step table plus the three non-obvious rules), and the three stale lines corrected.
- [x] **Public wall discoverable before its first publish** — created private always; visibility applied by `updateSprayWall` after the `PUBLISHED` latch.
- [x] **Orphan wall from an abandoned run** — the flow opens by offering an unfinished wall back, or discarding it.

### Answered rather than changed

- **`BoardVisibilityFields` mounts `useDeviceLocation` unconditionally.** Correct, and left as is: the hook requests no permission on mount, and `location.coords` stays null until the climber taps "Use my location" — the same contract `BoardForm` relied on behind its disclosure. Gating it on a disclosure the wizard does not have would mean a second code path for no behavioural difference.
- **A wall row can exist before any version** (raised as a note, not a bug). True by construction — the photo handler authorises against the wall — and that is exactly what the resume check now exists for.
- **A retry after a *non-committed* `createSprayWallVersion` re-uploads the photo.** Unavoidable: nothing distinguishes "did not commit" from "committed, response lost" except asking, and when the ask succeeds `planUploadRetry` adopts. The worst case is one unreferenced object, which the SW-17 cleanup job sweeps — it is never a row anybody can see.
- **No render smoke test for `SprayWallWizardScreen`.** Agreed and deferred to its own follow-up, at the coordinator's call: it needs the boards-index-sized mock set and is better landed on its own than bolted onto a fix commit.

### Gates at `3473acd7f`

`vp run typecheck:mobile` clean · `spray` suites 20 files / **305 tests** · `boards` 29 files / **407 tests** · `vp check` 0 errors, 370 warnings (baseline).

**Comment by marcodejongh** [comment]
Review items addressed (claude[bot] round at 15c5db351, verdict ready): the three observations are non-blocking and stay as noted — the upload-retry invariant test is tracked with the wizard render test in #5497; the `beforeRemove` cast is guarded and matches the existing pattern; orphaned draft walls are swept by SW-17's purge (#5498). No code change.

**Comment by marcodejongh** [comment]
Triage of the claude[bot] round at `15c5db351` (the "two correctness bugs" verdict). Both read the code wrong; neither reproduces.

**"Stuck spinner after publish + bind failure" — not a bug.** The claim is that `PUBLISH_FAILED` leaves `step` at `done`. It does not: the reducer sets `step: 'publish'` unconditionally (`add-wall-machine.ts:414-418`), which is the whole point of latching `published` apart from the step. `add-wall-machine.test.ts:416` asserts exactly this sequence — `PUBLISHED` (step `done`) then `PUBLISH_FAILED` gives `step === 'publish'`, `published === true`, and the error on screen with a retry that binds and does not re-publish.

**"Upload retry guard skipped on first retry" — not a bug.** The claim is that `attempts === 0` on a first retry. The first `UPLOAD_STARTED` already incremented it to 1 (`add-wall-machine.ts:333`), so a retry reads `attempts === 1` and `planUploadRetry` runs. The suggested fix — query before every upload — is the thing `planUploadRetry`'s own comment rejects: a first attempt has nothing to reconcile against.

There is a narrower real edge behind it, kept as-is: `PHOTO_PICKED` resets `attempts` to 0 while the wall row persists, so backing out of a failed upload and choosing a *different* photo skips the reconcile on that one attempt. It self-heals — `createSprayWallVersion` refuses on the open draft, the climber sees the server's message, and the next retry adopts — at a cost of one unreferenced object that SW-17 sweeps. Not worth a round trip on every first upload.

**Minor items:** `cornerLabels` / `seedKey` allocations are per-render in a non-list gesture component, below the CLAUDE.md checklist's bar; `resume-draft.ts`'s `status.toLowerCase() === 'draft'` is deliberately more tolerant than an enum compare, not less; clearing the gym leaving `locationName` matches `useBoardBuilder`'s existing behaviour and is the field the climber can edit.

**Changed in this pass:** `retryResume` now depends on React Query's stable `refetch` instead of the whole result object (an unaddressed item from the earlier round), and the test plan now starts by enabling the `spray-walls` flag and drags the angle slider rather than typing into it.

Gates green at `d23b55c09`: `vp run typecheck:mobile` clean, `vp run test:mobile` 936 files / 10,495 tests, `vp run check:mobile-bundle` OK, `vp check` 0 errors / 370 warnings (baseline).
